translator/source_code/combine_wzdx.py

```diff
@@ -0,0 +1,157 @@
+import json
+import copy
+from shapely.geometry.polygon import Polygon
+from shapely.geometry import Point
+import geopy
+from geopy.distance import geodesic
+import pyproj
+import getopt
+import sys
+
+
+def main():
+
+    icone_file, cotrip_file, outputfile = parse_arguments(
+        sys.argv[1:], default_output_file_name='combined_wzdx_message.geojson')
+
+    with open(outputfile, 'w+') as f:
+        wzdx_icone = json.loads(open(
+            icone_file, 'r').read())
+        wzdx_cotrip = json.loads(open(
+            cotrip_file, 'r').read())
+
+        polygon = generate_polygon(
+            wzdx_cotrip['features'][0]['geometry']['coordinates'], 100)
+        feature = iterate_feature(polygon, wzdx_icone)
+        if feature:
+
+            f.write(json.dumps(combine_wzdx(
+                wzdx_cotrip, wzdx_icone, feature), indent=2))
+            print('duplicate iCone and COtrip messages were found and combined WZDx message was written to combined_wzdx_message.geojson file')
+
+        else:
+            print('no duplicate messages were found')
+
+
+def combine_wzdx(wzdx_cotrip, wzdx_icone, icone_feature):
+    combined_wzdx = copy.deepcopy(wzdx_cotrip)
+    combined_wzdx['features'][0]['properties']['vehicle_impact'] = icone_feature['properties']['vehicle_impact']
```

Comment:
Is this `vehicle_impact` value guaranteed to be in both iCone and COtrip data?

Reply:
@abinashkonersman What I mean here is we are assigning the `vehicle_impact` value from the `icone_feature` object. Is that object guaranteed to have a "properties" property with a "vehicle_impact" property or is it possible for that property to be missing?

Likewise, we're looking to the first feature in the `combined_wzdx` object. Is that object guaranteed to have a list of features, each with "properties" and "vehicle_impact" properties? We likely need to add a check here to ensure this data structure and object properties exist  before assigning to them to prevent any issues in the future.

Reply:
Each **feature** is required to have properties, and **properties** is required to have **vehicle_impact**. It is however theoretically possible for a WZDx message to have an empty set of features, this should be corrected.

Reply:
should be fixed